checker/check_runner.go

```diff
@@ -96,21 +95,23 @@ func (r *Runner) Run(ctx context.Context, c Check) CheckResult {
 				fmt.Sprintf("requiredType: %s not supported by check %s", fmt.Sprint(unsupported), r.CheckName)))
 	}
 
-	l := NewLogger()
-	ctx, err := tag.New(ctx, tag.Upsert(stats.CheckName, r.CheckName))
+	err := stats.InitMetrics()
+	if err != nil {
+		panic(err)
+	}
+
 	if err != nil {
 		l.Warn(&LogMessage{Text: fmt.Sprintf("tag.New: %v", err)})
 	}
 
-	ctx, err = tag.New(ctx, tag.Upsert(stats.RepoHost, r.CheckRequest.Repo.Host()))
+	ctx = context.WithValue(ctx, stats.RepoHost, r.CheckRequest.Repo.Host())
```

Comment:
does it make a difference when you pass the value as an attribute, vs in the context? You do both in this PR.

 Skimming one migration page, I thought otel didnt support context based attributes?